src/main/java/bio/terra/cli/command/Main.java

```diff
@@ -61,18 +72,61 @@ public static void main(String... args) {
   public void run() {}
 
   /**
-   * Custom exception handler class that suppresses the stack trace when an exception is thrown.
-   * Instead, it just prints the exception message and exits the process.
+   * Custom handler class that intercepts all exceptions.
+   *
+   * <p>There are three categories of exceptions. All print a message to stderr and log the
+   * exception.
+   *
+   * <p>- User-facing = user can fix
+   *
+   * <p>- Internal = user cannot fix, exception specifically thrown by CLI code
+   *
+   * <p>- Unexpected = user cannot fix, exception not thrown by CLI code
+   *
+   * <p>The internal and unexpected cases are very similar, except that the message on the internal
+   * exception might be more readable/relevant.
    */
-  private static class PrintExceptionMessageHandler
+  private static class UserFacingExceptionHandler
       implements CommandLine.IExecutionExceptionHandler {
+
+    @Override
     public int handleExecutionException(Exception ex, CommandLine cmd, ParseResult parseResult) {
+      String errorMessage;
+      boolean printPointerToLogFile;
+      if (ex instanceof UserFacingException) {
+        errorMessage = ex.getMessage();
+        printPointerToLogFile = false;
+      } else if (ex instanceof InternalErrorException) {
+        errorMessage =
+            ex.getMessage() + (ex.getCause() != null ? ": " + ex.getCause().getMessage() : "");
+        printPointerToLogFile = true;
+      } else {
+        errorMessage = ex.getClass().getCanonicalName() + ": " + ex.getMessage();
```

Comment:
Potential formatting tweak to give users a bit more context:

"An unexpected error occurred in {className}: {message}"

Reply:
Changed.